okio/src/test/kotlin/okio/ByteStringTest.kt

```diff
@@ -16,21 +16,67 @@
 
 package okio
 
+import okio.ByteString.Companion.decodeBase64
 import okio.ByteString.Companion.decodeHex
+import okio.ByteString.Companion.encodeUtf8
+import okio.internal.commonAsUtf8ToByteArray
 import kotlin.test.Test
 import kotlin.test.assertEquals
 import kotlin.test.assertFalse
+import kotlin.test.assertNotEquals
+import kotlin.test.assertSame
 import kotlin.test.assertTrue
+import kotlin.test.fail
+
+class CommonByteStringTest {
+  class Platform : AbstractByteStringTest(ByteStringFactory.PLATFORM)
+  class Okio : AbstractByteStringTest(ByteStringFactory.OKIO)
+}
+
+interface ByteStringFactory {
+  fun decodeHex(hex: String): ByteString
```

Comment:
why is this in the factory? It’s the same for both implementations?

Reply:
This file is an attempt to recreate the ByteStringTest class from the JVM project. When/if SegmentedByteString moves to common there will be cases where we will need this method. (It also just made it easier when I copy and pasted the unit tests... 😄)